scripts/cfos/make_dataset_data_tables.py

```diff
@@ -143,57 +108,98 @@ def load_excel_input(excel, expected_dataset_cols):
+    return schema_dict, column_dict
 
 
 if __name__ == "__main__":
 
     parser = argparse.ArgumentParser(description='Create, validate, and upload (optionally ) Terra dataset specific data tables to a Terra workspace.')
 
-    parser.add_argument('-d', '--dataset_name', required=True, type=str, help='Dataset type to apply validations and data table structure. ex. proteomics, atac-seq, transcriptomics, singe-cell')
-    parser.add_argument('-x', '--excel', required=True, type=str, help='CFoS data excel file (.xlsx) - based on CFoS data intake template file.')
-    parser.add_argument('-j', '--schema', required=True, type=str, help='Dataset schema file (.json) - defines per dataset tables and associated attributes.')
-    parser.add_argument('-p', '--project', required=False, type=str, help='Terra workspace project (namespace).')
-    parser.add_argument('-w', '--workspace', required=False, type=str, help='Terra workspace name.')
-    parser.add_argument('-v', '--validate', required=False, action='store_true', help='Set parameter to run only validation on input excel file.')
-    parser.add_argument('-u', '--upload', required=False, action='store_true', help='Set parameter to upload data table files to workspace. default = no upload')
+    # REK TODO: update so JSON is assumed to live in an internal directory location and not as an input file. 
+    # Goal: allow qualified people to make easy updates to schemas, but don't encourage random users to play with it.
+
+    parser.add_argument(
+        '-d', 
+        '--dataset_name', 
+        required=False,
+        type=str, 
+        help='Dataset type to apply validations and data table structure. ex. proteomics, atac-seq, transcriptomics, singe-cell'
+    )
+    parser.add_argument(
+        '-x', 
+        '--excel', 
+        required=False,
+        type=str, 
+        help='CFoS data excel file (.xlsx) - based on CFoS data intake template file.'
+    )
+    parser.add_argument(
+        '-p', 
+        '--project', 
+        required=False, 
+        type=str, 
+        help='Terra workspace project (namespace).'
+    )
+    parser.add_argument(
+        '-w', 
+        '--workspace', 
+        required=False, 
+        type=str, 
+        help='Terra workspace name.'
+    )
+    parser.add_argument(
+        '-v', 
+        '--validate', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to run only validation on input excel file.'
+    )
+    parser.add_argument(
+        '-u', 
+        '--upload', 
+        required=False, 
+        action='store_true', 
+        help='Set parameter to upload data table files to workspace. default = no upload'
+    )
 
     args = parser.parse_args()
+    schema_json = 'dataset_tables_schema.json'
+
+    # TODO: Validate that schema exists; error and quit if not. Store validation code in new requirements or file_paths validation file.
+    # ask John - why write separate validation? Why not just use a Try block? 
```

Comment:
philosophical: also, why not just allow the error in the first place? if the schema doesn't exist, what happens? is whatever native error happens sufficient? what's the benefit of adding extensive checks here?

Reply:
John requested I add this error when reviewing the code. I'm not sure why it matters whether it's caught or not given the level of error handling elsewhere in the code. It needs to be a fatal error regardless, and the code won't execute without the schema either way.

Reply:
I think it is in general better practice to have try/except blocks when you can, but in this case I don't think the user will be impacted much either way.

Reply:
Pinged @kilonzi on slack asking about this again. I think it may be fine to remove this/not include it.

Reply:
Good catch here @mmorgantaylor, this is probably outdated but we can remove try/catch. A more general way to look at it is through [LBYL](https://docs.python.org/3.5/glossary.html#term-lbyl) and [EAFP](https://docs.python.org/3.5/glossary.html#term-eafp). We removing it here because it provided no additional benefit, other than fail and we do not anticipate to handle the exception in any way.

Reply:
I think I've removed this and the comments now.